text/001x-battery_improvements.md

```diff
@@ -0,0 +1,286 @@
+      <description>Battery status flags for fault, health and state indication.</description>
+      <entry value="1" name="MAV_BATTERY_STATUS_FLAGS_READY_TO_USE">
+        <description>
+          The battery is ready to use (fly).
+          Only set if the battery is safe and ready to fly with (not charging, no critical faults, such as those that would set MAV_BATTERY_STATUS_FLAGS_REQUIRES_SERVICE or MAV_BATTERY_STATUS_FLAGS_BAD_BATTERY, etc.).
+        </description>
+      </entry>
+      <entry value="2" name="MAV_BATTERY_STATUS_FLAGS_CHARGING">
+        <description>
+          Battery is charging.
+          Not ready to use (MAV_BATTERY_STATUS_FLAGS_READY_TO_USE would not be set).
+        </description>
+      </entry>
+      <entry value="4" name="MAV_BATTERY_STATUS_FLAGS_FAULT_CELL_BALANCING">
+        <description>
+          Battery is cell balancing (during charging).
+          Not ready to use (MAV_BATTERY_STATUS_FLAGS_READY_TO_USE would not be set).
+        </description>
+      </entry>
+      <entry value="8" name="MAV_BATTERY_STATUS_FLAGS_FAULT_CELL_IMBALANCE">
+        <description>
+          Battery cells are not balanced.
+          Not ready to use.
+        </description>
+      </entry>
+      <entry value="16" name="MAV_BATTERY_STATUS_FLAGS_AUTO_DISCHARGING">
+        <description>
+          Battery is auto discharging (towards storage level).
+          Not ready to use (MAV_BATTERY_STATUS_FLAGS_READY_TO_USE would not be set).
+        </description>
+      </entry>
+      <entry value="32" name="MAV_BATTERY_STATUS_FLAGS_REQUIRES_SERVICE">
+        <description>
+          Battery requires service (not safe to fly). 
+          This is set at vendor discretion.
+          It is likely to be set for most faults, and may also be set according to a maintenance schedule (such as age, or number of recharge cycles, etc.).
+        </description>
+      </entry>
+      <entry value="64" name="MAV_BATTERY_STATUS_FLAGS_BAD_BATTERY">
+        <description>
+          Battery is faulty and cannot be repaired (not safe to fly). 
+          This is set at vendor discretion.
+          The battery should be disposed of safely.
+        </description>
+      </entry>
+      <entry value="128" name="MAV_BATTERY_STATUS_FLAGS_PROTECTIONS_ENABLED">
+        <description>
+          Automatic battery protection monitoring is enabled. 
+          When enabled, the system will monitor for certain kinds of faults, such as cells being over-voltage.
+          If a fault is triggered then and protections are enabled then a safety fault (MAV_BATTERY_STATUS_FLAGS_FAULT_PROTECTION_SYSTEM) will be set and power from the battery will be stopped.
+          Note that the associated fault (such as MAV_BATTERY_STATUS_FLAGS_FAULT_OVER_VOLT) should always be set whether or not the protection system is engaged.
+        </description>
+      </entry>
+      <entry value="256" name="MAV_BATTERY_STATUS_FLAGS_FAULT_PROTECTION_SYSTEM">
```

Comment:
oh you can ignore my other reply, I didn't realize this was a new flag separate from MAV_BATTERY_STATUS_FLAGS_PROTECTIONS_ENABLED

Reply:
@dakejahl Just to be very clear, there are two ways this might work. I think your system does the first one, but the second seems a little more flexible. 

Currently the RFC says:
- MAV_BATTERY_STATUS_FLAGS_PROTECTIONS_ENABLED - actively monitoring for some fault. Turn off if we get one.
- MAV_BATTERY_STATUS_FLAGS_FAULT_PROTECTION_SYSTEM - the fault protections triggered because we got a fault. 

But (with slightly modified names) it could be:
- MAV_BATTERY_STATUS_FLAGS_PROTECTIONS_ENABLED - actively monitoring for some fault. Engage if we get a fault OR if we turn choose to engage the protections.
- MAV_BATTERY_STATUS_FLAGS_PROTECTION_ENGAGED - the fault protections are engaged so there is no power. You would get this if you are enabled and there is a fault, OR if you decided you just wanted to engage the protections.

So what's the difference?
- The first case implies that you only turn off for a fault, while the second implies you can explicitly turn on the protections (turn off power). If you got not other faults going high, you would assume this was just engaged deliberately.

Does what I am saying make sense? If so, do you agree the second set of names are better?

Reply:
Ping @dakejahl 
FYI, moving this out of draft for wider discussion. Would like to confirm the above.

Reply:
>Does what I am saying make sense? If so, do you agree the second set of names are better?

I think the first names are better. The reason being is that the fault protection system is just an automatic way to turn off the output FETs. The key thing is it's an automatic action.The output FETs could be turned off manually if you wanted, and that could be reflected in the (now removed) ACTIVE flag. I'm not sure if a flag indicating that the output FETs are enabled/disabled is very useful, but that was the original reason behind the ACTIVE flag.